DemoApp/Sources/Components/Stats/ParticipantStatsViewModel.swift

```diff
@@ -0,0 +1,102 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import StreamVideo
+import StreamVideoSwiftUI
+import SwiftUI
+
+@MainActor
+class ParticipantStatsViewModel: ObservableObject {
+    
+    let call: Call
+    let participant: CallParticipant
+    
+    private var cancellables = Set<AnyCancellable>()
+    
+    @Published var statsEntries = [StatsEntry]()
+    @Published var allStatsShown = false {
+        didSet {
+            if allStatsShown && statsReport == nil {
+                statsReport = call.state.statsReport
+            } else if !allStatsShown {
+                statsReport = nil
+            }
+        }
+    }
+    
+    var statsReport: CallStatsReport?
+    
+    init(
+        call: Call,
+        participant: CallParticipant
+    ) {
+        self.call = call
+        self.participant = participant
+        self.subscribeToStatsUpdates()
+    }
+
+    private func subscribeToStatsUpdates() {
+        call.state.$statsReport.sink { [weak self] statsReport in
```

Comment:
Can we move this one in background with `receive(on: DispatchQueue.global())` as the calculation isn't really required to be on MainThread (UI changes can be moved to Main when necessary)

Reply:
hmm, there's no calculation here, just setting the values. The calculation is already done on a background thread in the LLC.